assembl/static2/js/app/components/debate/thread/answerForm.jsx

```diff
@@ -0,0 +1,136 @@
+import React from 'react';
+import { PropTypes } from 'prop-types';
+import { connect } from 'react-redux';
+import { graphql } from 'react-apollo';
+import { Row, Col, FormGroup, Button } from 'react-bootstrap';
+import { Translate, I18n } from 'react-redux-i18n';
+
+import createPostMutation from '../../../graphql/mutations/createPost.graphql';
+import { updateAnswerPostBody, updateAnswerPostBodyRemaingChars, updateActiveAnswerFormId } from '../../../actions/postsActions';
+import { displayModal, displayAlert } from '../../../utils/utilityManager';
+import { getConnectedUserId } from '../../../utils/globalFunctions';
+import { getCurrentView, getContextual } from '../../../utils/routeMap';
+import { TxtAreaWithRemainingChars } from '../../common/txtAreaWithRemainingChars';
+
+const TEXT_AREA_MAX_LENGTH = 3000;
+const TEXT_AREA_ROWS = 10;
+
+const AnswerForm = ({
+  body,
+  bodyRemainingChars,
+  mutate,
+  updateBody,
+  updateBodyChars,
+  slug,
+  parentId,
+  ideaId,
+  refetchIdea,
+  hideAnswerForm
+}) => {
+  const handleBodyChange = (e) => {
+    const maxChars = TEXT_AREA_MAX_LENGTH;
+    const length = e.target.value.length;
+    const remaining = maxChars - length;
+    updateBodyChars(remaining);
+    updateBody(e.target.value);
+  };
+
+  const resetForm = () => {
+    hideAnswerForm();
+    updateBodyChars(TEXT_AREA_MAX_LENGTH);
+    updateBody('');
+  };
+
+  const inviteToLogin = () => {
```

Comment:
I would like to dispatch an action and store if the user is connected. Then I should move this method in the `main.jsx`. We need to display this modal at several places in the app. What do you think?